source/common/config/grpc_subscription_impl.h

```diff
@@ -21,27 +21,28 @@ class GrpcSubscriptionImpl : public Config::Subscription {
                        Stats::Scope& scope, const RateLimitSettings& rate_limit_settings,
                        std::chrono::milliseconds init_fetch_timeout, bool skip_subsequent_node)
       : callbacks_(callbacks),
-        grpc_mux_(local_info, std::move(async_client), dispatcher, service_method, random, scope,
-                  rate_limit_settings, skip_subsequent_node),
+        grpc_mux_(std::make_shared<Config::GrpcMuxImpl>(local_info, std::move(async_client),
+                                                        dispatcher, service_method, random, scope,
+                                                        rate_limit_settings, skip_subsequent_node)),
         grpc_mux_subscription_(grpc_mux_, callbacks_, stats, type_url, dispatcher,
                                init_fetch_timeout) {}
 
   // Config::Subscription
   void start(const std::set<std::string>& resource_names) override {
     // Subscribe first, so we get failure callbacks if grpc_mux_.start() fails.
     grpc_mux_subscription_.start(resource_names);
-    grpc_mux_.start();
+    grpc_mux_->start();
   }
 
-  void updateResources(const std::set<std::string>& update_to_these_names) override {
-    grpc_mux_subscription_.updateResources(update_to_these_names);
+  void updateResourceInterest(const std::set<std::string>& update_to_these_names) override {
+    grpc_mux_subscription_.updateResourceInterest(update_to_these_names);
   }
 
-  GrpcMuxImpl& grpcMux() { return grpc_mux_; }
+  std::shared_ptr<Config::GrpcMuxImpl> grpcMux() { return grpc_mux_; }
```

Comment:
the return type should be `const std::shared_ptr<Config::GrpcMuxImpl>&` to avoid copy shared_ptr

Reply:
I think that's probably right, but I can't immediately be 100% sure it's safe. This class is going to be entirely replaced anyways, so I would say it's safer and simpler to leave this alone.